napari/layers/base/base.py

```diff
@@ -222,7 +221,7 @@ def __init__(
         super().__init__()
 
         if name is None and data is not None:
-            name = magic_name(data, path_prefix=ROOT_DIR)
+            name = magic_name(data)
```

Comment:
@tlambert03 can you comment on this change? Not sure why it's in this PR...

Reply:
Can you reread the end of the post in https://github.com/napari/napari/pull/2683 and see if it makes sense?

Reply:
Does this actually address that comment? I thought the point with ROOT_DIR was just to avoid naming it something from the napari code hierarchy, including generated code?

Reply:
(in other words: I see now that MultiScaleData messes with magic_name, but I don't see whether this does anything about that and I suspect it might reintroduce other bugs?)

Reply:
Ah, sorry, I thought you were generally asking what this had to do with magic_name.

The only change here is to make root_dir the _default_ value on magic_name (see change in magic_name definition too). (It's pretty much the only valid value for path_prefix anyway).  And I don't recall there being anywhere else in the code that calls magic_name without that  root_dir as the value.